cirq/circuits/circuit_test.py

```diff
@@ -3056,3 +3056,16 @@ def test_moment_groups():
 (0, 7): ────H──────H─────────────────────
            └──┘   └───┘   └───┘   └──┘
 """, use_unicode_characters=True)
+
+
+def test_from_qasm():
```

Comment:
You should add a consistency test asserting that the QASM we produce can be parsed.

Ideally it would be the case that parse(x) == parse(produce(parse(x))).

Reply:
That should be possible on the unitary level, is that what you're thinking ? It will be more challenging on the actual gate level - we'll have to implement the inverse of the `__decompose__` functions in an unambiguous way for that.